sdk/core/azure-core/test/ut/transport_adapter_base_test.cpp

```diff
@@ -588,15 +588,10 @@ namespace Azure { namespace Core { namespace Test {
       int64_t size,
       std::string expectedBody)
   {
+    (void)size;
     auto body = response.ExtractBodyStream();
     EXPECT_EQ(body, nullptr);
     std::vector<uint8_t> bodyVector = response.GetBody();
-    int64_t bodySize = bodyVector.size();
-
-    if (size > 0)
-    { // only for known body size
-      EXPECT_EQ(bodySize, size);
-    }
```

Comment:
@vhvb1989 as discussed offline removing this check probably doesn't drop any coverage, but does allow for better diagnostic logging if the assert below fails, during test runs.

Up to you to take such a change after verifying if we can safely remove this assert and rely on EXPECT_STREQ.

This draft PR was just meant to highlight the potential benefit of writing tests in a way that causes of failures are easier to understand from the logs, especially for intermittent issues.